### ISSUE TITLE
Refactor all BbPromise.x usage which doesn't have native counterparts

### ISSUE THREAD

**Issue by medikoo** [serverless/serverless#8369]
### Use case description

Partially covers https://github.com/serverless/serverless/issues/7171

Should be addressed after https://github.com/serverless/serverless/issues/8368

Bluebird exposes a lot of functions and methods that have no standard counterparts, we should refactor it to native promises syntax

### Proposed solution

Refactor to async/await  all functions that contain usage of following methods:

_Note: To avoid large, problematic to handle PR's. let's replace each function with different PR_

- [ ] `BbPromise.bind` - Most likely, in all used cases, bind wrapper can be just removed (no counterpart needed)
- [x] `BbPromise.delay` - Timeout could be implemented with help of `timers-ext/promise/sleep`
- [ ] `BbPromise.each` - We can simply use `for (..) { await .. }` construct
- [x] `BbPromise.fromCallback` - Use case for which it's used, it's probably no longer actual (if it is, we need to come up with `new Promise(..)` construct)
- [x] `BbPromise.join` - Replace with `Promise.all`
- [ ] `BbPromise.map` - Replace with `Promise.all`
- [ ] `BbPromise.mapSeries` - We can simply use `for (..) { await .. }` construct
- [x] `BbPromise.promisify` - When `fs` functions are wrapped, resort to `fs.promises`. In other cases use `require('util').promisify`
- [ ] `BbPromise.promisifyAll` - When `fs` functions are wrapped, resort to `fs.promises`. In other cases use `require('util').promisify`
- [x] `BbPromise.props` - Refactor with `Promise.all`
- [x] `BbPromise.reduce` - We can simply use `for (..) { await .. }` construct
- [ ] `BbPromise.try` - As all functions containing it are now declared as async, this wrapper can simply be just removed

**Comment by ifitzsimmons** [comment]
I created a branch to start chipping away at this. I refactored everything under `lib/plugins/aws/deploy`. Link to branch is [here](https://github.com/ifitzsimmons/serverless/tree/8369-refactor-bbpromise). 

I can submit the pull request once my PR for #8368 has been merged. I used the `async` tagging changes from that branch while refactoring. I'll refactor some of the other folders tomorrow.

**Comment by medikoo** [comment]
@ifitzsimmons great thanks for help! Ideally if we address it step by step, as outlined in issues. First by addressing #8368, and we have a lot to cover over there. Have you checked that issue?

**Comment by ifitzsimmons** [comment]
@medikoo I did. I made the async tagging updates for #8368 and am just waiting for it to be merged. Do you want to complete ALL of #8368 before refactoring the `Promise` code, or can we start refactoring the `Promise`s to `async/await` for a folder once we tag the `Promise`-based functions with `async` and merge it in?

**Comment by medikoo** [comment]
> Do you want to complete ALL of #8368 

Yes, ideally if that step is finalized entirely before moving with next steps

**Comment by duytrandev** [comment]
Create #11797 to refactor `BbPromise.each()`
I don't see these being use:
`BbPromise.delay`
`BbPromise.fromCallback`
`BbPromise.join`
`BbPromise.promisify`
`BbPromise.props`
`BbPromise.reduce`

So we can safely cross them out in the original comment

**Comment by medikoo** [comment]
@sleepwithcoffee thanks, I've updated the description. Concerning this scope of work, ideally if first we cover https://github.com/serverless/serverless/issues/8368. Intentionally this async refactor, was planned into multiple carefully ordered steps - so we do not introduce some regressions (not all of the affected parts are well tested)

**Comment by sleepwithcoffee** [comment]
hi @medikoo I think this was closed by accident, let's reopen this ticket
